doc/about.rst

```diff
@@ -0,0 +1,26 @@
+
+About
```

Comment:
I wonder if we should separate into two files `About` and `Related projects`?
Not a big thing but if we stuck them together it's ignore by the `Next` button:
![image](https://github.com/skrub-data/skrub/assets/62058944/cbe81feb-8164-4627-9e3d-a8df358f7ce3)
Good to know for future pages, they should each be done separately.